File: src/typescript/src/models/PortAndRouteInformation.ts
/**
 * Represents the Port and Route Information section of SVD.
 */
export type PortAndRouteInformation = {
    DeparturePort: string
    DepartureTime: Date
    DestinationPort: string
    Etapbp: Date
    EtaBerth: Date
    Etavts: Date
    DistanceToNextWaypointNm: number
    DistanceToNextWaypointKm: number
    DistanceToDestinationPortNm: number
    DistanceToDestinationPortKm: number
    ProjectedSpeedKn: number
    ProjectedSpeedKmh: number
    VoyageTimeMs: number
}

File: src/dotnet/src/StandardisedVesselDataset/Models/PortAndRouteInformation.cs
namespace StandardisedVesselDataset.Models;

/// <summary>
/// Represent the Port and Route Information section of SVD
/// </summary>
public class PortAndRouteInformation
{
    public string DeparturePort { get; set; }
    public DateTime DepartureTime { get; set; }
    public string DestinationPort { get; set; }
    public DateTime Etapbp { get; set; }
    public DateTime EtaBerth { get; set; }
    public DateTime Etavts { get; set; }
    public double DistanceToNextWaypointNm { get; set; }
    public double DistanceToNextWaypointKm { get; set; }
    public double DistanceToDestinationPortNm { get; set; }
    public double DistanceToDestinationPortKm { get; set; }
    public double ProjectedSpeedKn { get; set; }
    public double ProjectedSpeedKmh { get; set; }
    public TimeSpan VoyageTime { get; set; }
}